### DISCUSSION TITLE
RECT with Structure

### DISCUSSION THREAD

**Discussion by hxhgxy** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#2839] [Q&A] [answered]
there is "rect with" structure in ShadowEdgeConverter:
`

    if (edges.HasFlag(ShadowEdges.Left))
    {
	    rect = rect with { X = -blurRadius, Width = width + blurRadius };
    }
    
    if (edges.HasFlag(ShadowEdges.Top))
    {
	    rect = rect with { Y = -blurRadius, Height = height + blurRadius };
    }
    
    if (edges.HasFlag(ShadowEdges.Right))
    {
	    rect = rect with { Width = rect.Width + blurRadius };
    }
    
    if (edges.HasFlag(ShadowEdges.Bottom))
    {
	    rect = rect with { Height = rect.Height + blurRadius };
    }
`

is it supported in Language version 10.0 above? What should to do in language version 9.0?

Thanks,
Tiger

**Comment by Keboo** [comment, marked as answer]
In prior versions of the language, the way to check for a given flag is with bitwise operators.

So:
```c#
if (edges.HasFlag(ShadowEdges.Left))
{
    ...
}
```

Becomes:
```c#
if ((edges & ShadowEdges.Left) == ShadowEdges.Left)
{
       ...         
}
```

**Comment by hxhgxy** [comment]
thanks.
